sdk/python/kfp/v2/components/experimental/component_spec.py

```diff
@@ -0,0 +1,360 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Definitions for component spec."""
+
+from dataclasses import dataclass
+from typing import Any, Mapping, Optional, Sequence, Union
+
+from kfp.components import _structures as v1_components
+from kfp.components import _data_passing
+
+
+@dataclass
+class InputSpec:
+  """Component input definitions.
+
+  Attributes:
+    name: The name of the input.
+    type: The type of the input.
+    default: Optional; the default value for the input.
+  """
+  name: str
+  type: Any
```

Comment:
Note - at some point, we want to formalize types in our SDK. At that point, we should update these. Just a thought.

Reply:
Ack.